observability-kit-starter/src/main/java/com/vaadin/observability/JsonNodeSpanWrapper.java

```diff
@@ -0,0 +1,276 @@
+                Map.Entry<AttributeKey, Object> entry = attributeValue(key,
+                        element);
+                if (entry != null) {
+                    values.add(entry.getValue());
+                    if (attributeKey == null) {
+                        // As per specs, the array MUST be homogeneous,
+                        // i.e., it MUST NOT contain values of different
+                        // types.
+                        attributeKey = entry.getKey();
+                    }
+                }
+            }
+            if (attributeKey == null) {
+                // Rejecting empty array, as the type can't be determined
+                return null;
+            }
+
+            switch (attributeKey.getType()) {
+            case LONG:
+                attributeKey = AttributeKey.longArrayKey(key);
+                break;
+            case DOUBLE:
+                attributeKey = AttributeKey.doubleArrayKey(key);
+                break;
+            case STRING:
+                attributeKey = AttributeKey.stringArrayKey(key);
+                break;
+            case BOOLEAN:
+                attributeKey = AttributeKey.booleanArrayKey(key);
+                break;
+            default:
+                // Arrays can contain only primitive values
+                attributeKey = null;
+                break;
+            }
+            if (attributeKey == null) {
+                return null;
+            }
+            return new AbstractMap.SimpleImmutableEntry<>(attributeKey,
+                    values);
+        }
+        return null;
+    }
+
+    public JsonNodeSpanWrapper(String frontendId, JsonNode resourceNode,
+            JsonNode scopeNode, JsonNode spanNode) {
+        this.spanContext = SpanContext.create(spanNode.get("traceId").asText(),
+                spanNode.get("spanId").asText(), TraceFlags.getDefault(),
+                TraceState.getDefault());
+
+        this.parentSpanContext = SpanContext.create(
+                spanNode.get("traceId").asText(),
+                spanNode.has("parentSpanId") ?
+                        spanNode.get("parentSpanId").asText() : null,
+                TraceFlags.getDefault(),
+                TraceState.getDefault());
+
+        ResourceBuilder resourceBuilder = Resource.builder();
+        this.resource = resourceBuilder
+                .putAll(getAttributes(resourceNode).build())
+                .build();
+
+        this.instrumentationLibraryInfo =
+                InstrumentationLibraryInfo.create(
+                        scopeNode.get("name").asText(),
+                        scopeNode.get("version").asText());
+
+        this.instrumentationScopeInfo = InstrumentationScopeInfo.create(
+                scopeNode.get("name").asText(),
+                scopeNode.get("version").asText(), null);
+
+        this.name = "Frontend: " + spanNode.get("name").asText();
+        this.kind = SpanKind.values()[spanNode.get("kind").asInt() + 1];
```

Comment:
The `kind`  value sent by the browser matches the index of the `SpanKind` enumeration, no need to add one unit
```suggestion
        this.kind = SpanKind.values()[spanNode.get("kind").asInt()];
```
Also, we may consider forcing it to `SpanKind.CLIENT` because if I recall correctly, most front-end instrumentation does not explicitly set the value, so we will get zero.